stage/pulumi_run_recorder.go

```diff
@@ -163,8 +163,50 @@ func (p *PulumiMySQLRunRecorder) findPulumiStackFromClusterFQDN(ctx context.Cont
 	return nil
 }
 
+func (p *PulumiMySQLRunRecorder) findStackFromClusterFQDN(ctx context.Context, clusterFQDN string) *PulumiResource {
+	// Check if the cluster FQDN matches the Presto DB pattern
+	if regexp.MustCompile(`.+\.ibm\.prestodb\.dev`).MatchString(clusterFQDN) {
+		return p.findPulumiStackFromClusterFQDN(ctx, clusterFQDN)
+	}
+
+	// Check if the cluster FQDN matches the blueray pattern
+	if regexp.MustCompile(`.+\.cloud\.ibm\.com`).MatchString(clusterFQDN) {
+		return p.findBluerayStackFromClusterFQDN(ctx, clusterFQDN)
+	}
+
+	// Log if the FQDN doesn't match any known pattern
+	log.Warn().Str("cluster_fqdn", clusterFQDN).Msg("cluster FQDN does not match any known pattern")
+	return nil
+}
+
+func (p *PulumiMySQLRunRecorder) findBluerayStackFromClusterFQDN(ctx context.Context, clusterFQDN string) *PulumiResource {
+	// Extract cluster name as the first part of the FQDN (before the first dot)
+	parts := regexp.MustCompile(`\.`).Split(clusterFQDN, 2)
```

Comment:
Here you do not need to use regex because you just need to get the part before the dot.
```suggestion
	parts := strings.SplitN(clusterFQDN, ".", 2)
```